src/panels/lovelace/cards/hui-area-card.ts

```diff
@@ -189,7 +195,7 @@ export class HuiAreaCard
             (entity) => entity.attributes.device_class === deviceClass
           )
         : entities
-    ).some(
+    ).find(
```

Comment:
Shouldn't this return all entities that are on?

Reply:
In the past we show a single icon if any boolean in that device_class was on. 

You're suggesting change to show multiple alert icons of the same type? Shouldn't be hard to do, just a diffrent behavior than originally designed.

Reply:
No, I think it should still be a single icon (we could show a badge on it if it where multiple though maybe)

Reply:
We now show the icon and more info of the first entity we find, without indication that there are more entities that are in alert state. I think that is weird, I think the dialog should show all entities that are in alert state.

Reply:
Ok, so instead of launching more-info, you want to show a new custom dialog that can show the state of multiple entities. 

Would you accept for now if we just drop the more-info, and then this feature just shows the device_class icon only, exactly like how it is currently handled? 

A new dialog might take some effort and time to create, and it's kind of a standalone new feeature I would think, doesn't need to be tied with this.

Reply:
Yep, def!

Reply:
Ok I nixed the more-info click action, now it behaves like before.